cli/tests/pcluster/api/controllers/test_cluster_instances_controller.py

```diff
@@ -5,43 +5,167 @@
+def cfn_describe_instances_mock_response(
+    instance_type="t2.micro", queue_name=None, node_type="HeadNode", cluster_name="clustername"
+):
+    instance_data = {
+        "InstanceId": "i-0a9342a0000000000",
+        "InstanceType": instance_type,
+        "KeyName": "jenkinsjun",
+        "LaunchTime": "2021-04-30T00:00:00+00:00",
+        "PrivateIpAddress": "10.0.0.79",
+        "PublicIpAddress": "1.2.3.4",
+        "State": {"Code": 16, "Name": "running"},
+        "Tags": [
+            {"Key": "parallelcluster:version", "Value": "3.0.0"},
+            {"Key": "parallelcluster:node-type", "Value": node_type},
+            {"Key": "parallelcluster:cluster-name", "Value": cluster_name},
+        ],
+    }
+    if queue_name:
+        instance_data["Tags"].append({"Key": "QueueName", "Value": queue_name})
+
+    return instance_data
+
+
+class TestDescribeClusterInstances:
+    url = "/v3/clusters/{cluster_name}/instances"
+    method = "GET"
+
+    def _send_test_request(
+        self, client, cluster_name="clustername", region="us-east-1", next_token=None, node_type=None, queue_name=None
+    ):
+        query_string = []
+        if region:
+            query_string.append(("region", region))
+        if next_token:
+            query_string.append(("nextToken", next_token))
+        if node_type:
+            query_string.append(("nodeType", node_type))
+        if queue_name:
+            query_string.append(("queueName", queue_name))
 
-    def test_delete_cluster_instances(self, client):
-        """Test case for delete_cluster_instances."""
-        query_string = [("region", "eu-west-1"), ("force", True)]
         headers = {
             "Accept": "application/json",
         }
-        response = client.open(
-            "/v3/clusters/{cluster_name}/instances".format(cluster_name="clustername"),
-            method="DELETE",
-            headers=headers,
-            query_string=query_string,
+        return client.open(
+            self.url.format(cluster_name=cluster_name), method=self.method, headers=headers, query_string=query_string
         )
-        assert_that(response.status_code).is_equal_to(204)
-
-    def test_describe_cluster_instances(self, client):
-        """Test case for describe_cluster_instances."""
-        query_string = [
-            ("region", "eu-west-1"),
-            ("nextToken", "next_token_example"),
-            ("nodeType", NodeType.HEAD),
-            ("queueName", "queue_name_example"),
-        ]
-        headers = {
-            "Accept": "application/json",
-        }
-        response = client.open(
-            "/v3/clusters/{cluster_name}/instances".format(cluster_name="clustername"),
-            method="GET",
-            headers=headers,
-            query_string=query_string,
+
+    @pytest.mark.parametrize(
+        "instances, params, expected_response",
+        [
+            ([], {}, {"instances": [], "nextToken": ""}),
```

Comment:
Can we assign None to nextToken in the response if we are not doing this already so that nextToken is not returned at all rather than being an empty string?
Ah probably the change is only required in your mock where you don't want to return a nextToken.

Also can you add a test that returns a string as a nextToken?